spec/spec_helper.rb

```diff
@@ -8,27 +8,58 @@
 require "pry"
 require "rspec"
 require "rdkafka"
+require "zlib"
+require "timeout"
 
-def rdkafka_config(config_overrides={})
-  config = {
+def rdkafka_base_config
+  {
     :"api.version.request" => false,
     :"broker.version.fallback" => "1.0",
     :"bootstrap.servers" => "localhost:9092",
-    :"group.id" => "ruby-test-#{Random.new.rand(0..1_000_000)}",
-    :"auto.offset.reset" => "earliest",
-    :"enable.partition.eof" => false
```

Comment:
It seems like there are two main changes in this PR-- the 60 second timeout in the `around(:each)`  on line 130, and the extraction of this configuration out of the default, therefore moving it out of the producer configuration.

Were these settings interfering with some of the producer-related tests?